lsp-mode.el

```diff
@@ -4245,6 +4289,17 @@ it has to calculate identation based on SRC block position."
             (with-no-warnings (undo-amalgamate-change-group change-group)))
           (progress-reporter-done reporter))))))
 
+(defun lsp--apply-additional-text-edits (edits &optional main-edit-region)
+  "Apply the additional EDITS described in TextEdit[] object.
+This will also verify if the additional text edits are applicable
+regarding to the MAIN-EDIT-REGION."
+  (unless (and main-edit-region
```

Comment:
main-edit-region seems to be redundant - additional text edits are valid as long as there is no edit by the user.

Reply:
The `main-edit-region` here is actually the edit region made by user. So as long as user doesn't edit overlappingly with additional texts, they will be applied

Reply:
> overlappingly

We are not looking for overlap but if there was a change after the first start point in the edits. Even if there is no overlap the edit will be invalid if there was change before it.

Reply:
Yes, the function is checking exactly for that.